deploy/sre-prometheus/management-cluster/100-dynatrace-workloads-monitoring.PrometheusRule.yaml

```diff
@@ -0,0 +1,91 @@
+      expr: |
+          sum(kube_node_status_condition{condition="Ready", status="false"}) == 0 and
+          sum(hypershift_hostedclusters) > 0 and (
+          kube_deployment_status_replicas{deployment="dynatrace-operator", namespace="dynatrace"} == 0 or
+          kube_deployment_status_replicas{deployment="dynatrace-webhook", namespace="dynatrace"} == 0 or
+          kube_deployment_status_replicas{deployment="opentelemetry-dynatrace-collector", namespace="dynatrace"} == 0 or
+          kube_statefulset_status_replicas{statefulset=~".*-activegate", namespace="dynatrace"} == 0 or
+          kube_daemonset_status_desired_number_scheduled{daemonset=~".*-oneagent", namespace="dynatrace"} == 0 )
+      for: 15m
+      labels:
+        namespace: dynatrace
+        severity: critical
+      annotations:
+        summary: "Dynatrace Workloads are failing on the Management Cluster"
+        description: "The Dynatrace deployment on the Management Cluster has failed to be deployed for 15 mins"
+
+  - name: sre-dynatrace-monitoring-stack-degraded
+    rules:
+    - alert: DynatraceOperatorDegradedSRE
+      expr: (sum(kube_deployment_spec_replicas{deployment="dynatrace-operator", namespace="dynatrace"}) without (deployment, instance, pod)) - (sum(kube_deployment_status_replicas_available{deployment="dynatrace-operator", namespace="dynatrace"}) without (deployment, instance, pod)) > 0
+      for: 10m
+      labels:
+        namespace: dynatrace
+        severity: critical
+      annotations:
+        summary: "The Dynatrace Operator is Degraded"
+        description: "{{$value}} pods are not running out of total pods for the Dynatrace Operator"
+  
+    - alert: DynatraceWebhookDegradedSRE
+      expr: (sum(kube_deployment_spec_replicas{deployment="dynatrace-webhook", namespace="dynatrace"}) without (deployment, instance, pod)) - (sum(kube_deployment_status_replicas_available{deployment="dynatrace-webhook", namespace="dynatrace"}) without (deployment, instance, pod)) > 0
+      for: 10m
+      labels:
+        namespace: dynatrace
+        severity: critical
+      annotations:
+        summary: "Dynatrace Webhook is Degraded"
+        description: "{{$value}} pods are not running out of total pods for the Dynatrace Webhook"
+
+    - alert: DynatraceOpenTelemetryCollectorDegradedSRE
+      expr: (sum(kube_deployment_spec_replicas{deployment="opentelemetry-dynatrace-collector", namespace="dynatrace"}) without (deployment, instance, pod)) - (sum(kube_deployment_status_replicas_available{deployment="opentelemetry-dynatrace-collector", namespace="dynatrace"}) without (deployment, instance, pod)) == 0
+      for: 15m
+      labels:
+        namespace: dynatrace
+        severity: critical
+      annotations:
+        summary: "Dynatrace OTEL Collector is Degraded"
+        description: "{{$value}} pods are not running out of total pods for the OpenTelemetry Dynatrace Collector"
+
+    - alert: DynatraceActiveGateDegradedSRE
```

Comment:
should this be critical? Do all instances need to be up at all times?

Reply:
Changed to warning severity